scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,285 @@
+import scipy.stats as stats
+import numpy as np
+from scipy._lib._util import check_random_state
+
+
+def _resample_data(data, n_resamples=None, random_state=None):
+    '''Resample the data'''
+    n = data.shape[-1]
+    if n_resamples:
+        # bootstrap - each row is a random resample of original observations
+        i = random_state.randint(0, n, (n_resamples, n))
+    else:
+        # jackknife - each row leaves out one observation
+        j = np.ones((n, n), dtype=bool)
+        np.fill_diagonal(j, False)
+        i = np.arange(n)
+        i = np.broadcast_to(i, (n, n))
+        i = i[j].reshape((n, n-1))
+    resamples = data[..., i]
+    return resamples
+
+
+def _percentile_of_score(data, score, axis):
+    '''Vectorized, simplified scipy.stats.percentileofscore'''
+    B = data.shape[axis]
+    return (data < score).sum(axis=axis) / B
+
+
+def _percentile_along_axis(theta_hat_b, alpha):
+    '''np.percentile with different percentile for each slice'''
+    shape = theta_hat_b.shape[:-1]
+    alpha = np.broadcast_to(alpha, shape)
+    percentiles = np.zeros_like(alpha)
```

Comment:
How do you want it changed? 
I think `alpha` is guaranteed to be a float in the current code because it comes from `confidence_interval`, which is explicitly `float(confidence_interval)`.